sql/expression/function/time.go

```diff
@@ -828,3 +895,78 @@ func weekFuncLogic(t time.Time) (interface{}, error) {
 	_, wk := t.ISOWeek()
 	return wk, nil
 }
+
+// TimeDiff subtracts the second argument from the first expressed as a time value.
+type TimeDiff struct {
+	expression.BinaryExpression
+}
+
+// NewTimeDiff creates a new NewTimeDiff expression.
+func NewTimeDiff(e1, e2 sql.Expression) sql.Expression {
+	return &TimeDiff{
+		expression.BinaryExpression{
+			Left:  e1,
+			Right: e2,
+		},
+	}
+}
+
+// Type implements the Expression interface.
+func (td *TimeDiff) Type() sql.Type { return sql.Time }
+
+// IsNullable implements the Expression interface.
+func (td *TimeDiff) IsNullable() bool { return false }
+
+func (td *TimeDiff) String() string {
+	return fmt.Sprintf("TIMEDIFF(%s, %s)", td.Left, td.Right)
+}
+
+// WithChildren implements the Expression interface.
+func (td *TimeDiff) WithChildren(children ...sql.Expression) (sql.Expression, error) {
+	if len(children) != 2 {
+		return nil, sql.ErrInvalidChildrenNumber.New(td, len(children), 2)
+	}
+	return NewTimeDiff(children[0], children[1]), nil
+}
+
+// Eval implements the Expression interface.
+func (td *TimeDiff) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	// check for valid types
+	if (td.Left.Type() == sql.Time || td.Left.Type() == sql.Timestamp ||
+		td.Left.Type() == sql.Datetime) && (td.Right.Type() == sql.Time ||
+		td.Right.Type() == sql.Timestamp || td.Right.Type() == sql.Datetime) {
+
+		// handle type mismatch
+		if td.Left.Type() != td.Right.Type() {
+			return nil, ErrArgumentTypeMisMatch.New("TIMEDIFF", td.Left, td.Right)
+		}
+
+		left, err := td.Left.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+
+		right, err := td.Right.Eval(ctx, row)
+		if err != nil {
+			return nil, err
+		}
+
+		if left == nil || right == nil {
+			return nil, ErrTimeUnexpectedlyNil.New("TIMEDIFF")
+		}
+
+		leftTime := time.Unix(0, left.(int64))
+		rightTime := time.Unix(0, right.(int64))
```

Comment:
so this works cause in my tests calling `time.UnixNano()` where calling `time.Unix()` breaks this...